rust/signed_doc/src/lib.rs

```diff
@@ -64,7 +65,55 @@ impl From<InnerCatalystSignedDocument> for CatalystSignedDocument {
     }
 }
 
+/// Catalyst Signed Document Error
+#[derive(Debug)]
+pub struct CatalystSignedDocError {
+    /// List of errors during processing.
+    report: ProblemReport,
+    /// Actual error.
+    error: anyhow::Error,
+}
+
+impl fmt::Display for CatalystSignedDocError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let report_json = serde_json::to_string(&self.report)
+            .unwrap_or_else(|_| String::from("Failed to serialize ProblemReport"));
+
+        write!(
+            fmt,
+            "CatalystSignedDocError {{ error: {}, report: {} }}",
+            self.error, report_json
+        )
+    }
+}
+
 impl CatalystSignedDocument {
+    /// Create a new Catalyst Signed Document from a COSE Sign document bytes.
+    ///
+    /// # Arguments
+    ///
+    /// * `cose_bytes` - COSE Sign document bytes.
+    ///
+    /// # Returns
+    ///
+    /// A new Catalyst Signed Document.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the COSE Sign document bytes are invalid or decode error.
+    pub fn new(cose_bytes: &[u8]) -> anyhow::Result<Self, CatalystSignedDocError> {
```

Comment:
For such simple `new` method with one argument, its better to implement `TryFrom<&[u8]>` trait.